### PR TITLE
DRAFT: Quantization configuration file

### PR DESCRIPTION
**Pull request by jinevening**
On going draft to support quantization config file.

ONE-DCO-1.0-Signed-off-by: Hyukjin Jeong <hj1.jeong@samsung.com>

---
Related to: https://github.com/Samsung/ONE/issues/8399

### CODE REVIEW

File: compiler/luci/pass/src/QuantizeWithMinMaxPass.cpp [hunk 1]
```diff
@@ -1665,6 +1665,17 @@ bool QuantizeWithMinMaxPass::run(loco::Graph *g)
...
+  for (auto layer_info : _ctx->layers_info)
+  {
+    auto name = layer_info.name;
+    auto dtype = layer_info.dtype == loco::DataType::S16 ? "int16" : "uint8";
+    auto granularity =
+      layer_info.granularity == QuantizationGranularity::ChannelWise ? "CWQ" : "LWQ";
+    std::cout << "name : " << name << std::endl;
+    std::cout << "dtype : " << dtype << std::endl;
+    std::cout << "granularity : " << granularity << std::endl;
+  }
```
Comment (jinevening) [inline compiler/luci/pass/src/QuantizeWithMinMaxPass.cpp:1677]: Input json file

```json
{
    "default_quantization_dtype" : "uint8",
    "default_granularity" : "channel",
    "layers" : [
        {
            "name" : "test",
            "dtype" : "int16",
            "granularity" : "channel"
        },
        {
            "name" : "test2",
            "dtype" : "uint8",
            "granularity" : "layer"
        }
    ]
}
```

Printed values
```
name : test
dtype : int16
granularity : CWQ
name : test2
dtype : uint8
granularity : LWQ
```

**Comment by seanshpark** [comment]
Q) is using JSON become mandatory or optional?

File: compiler/circle-quantizer/CMakeLists.txt [hunk 2]
```diff
@@ -1,6 +1,15 @@
+nnas_find_package(Jsoncpp)
```
[the remark this answers is not in the document]
Comment (jinevening) [reply compiler/circle-quantizer/CMakeLists.txt:1]: Ah, I forgot to upload the cmake files for jsoncpp. Thanks.

**Comment by jinevening** [comment]
> is using JSON become mandatory or optional?

I think it should be optional for backward-compatibility.

**Pull request closed without merging**